packages/@aws-cdk/aws-ec2/lib/network-acl.ts

```diff
@@ -0,0 +1,363 @@
+import { AclCidr, AclTraffic } from './network-acl-types';
+import { ISubnet, IVpc, SubnetSelection } from './vpc';
+
+/**
+ * A NetworkAcl
+ *
+ * @experimental
+ */
+export interface INetworkAcl extends IResource {
+  /**
+   * ID for the current Network ACL
+   * @attribute
+   */
+  readonly networkAclId: string;
+}
+
+/**
+ * A NetworkAclBase that is not created in this template
+ *
+ * @experimental
+ */
+abstract class NetworkAclBase extends Resource implements INetworkAcl {
+  public abstract readonly networkAclId: string;
+}
+
+/**
+ * Properties to create NetworkAcl
+ *
+ * @experimental
+ */
+export interface NetworkAclProps {
+  /**
+   * The name of the NetworkAcl.
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a networkAclName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly networkAclName?: string;
+
+  /**
+   * The VPC in which to create the NetworkACL.
+   */
+  readonly vpc: IVpc;
+
+  /**
+   * Subnets in the given VPC to associate the ACL with
+   *
+   * More subnets can always be added later by calling
+   * `associateWithSubnets()`.
+   *
+   * @default - No subnets associated
+   */
+  readonly subnetSelection?: SubnetSelection;
+}
+
+/**
+ * Define a new network ACL
+ *
+ * @experimental
+ */
+export class NetworkAcl extends NetworkAclBase {
+  /**
+   * Import an existing NetworkAcl into this app.
+   */
+  public static fromNetworkAclId(scope: Construct, id: string, networkAclId: string): INetworkAcl {
+    class Import extends NetworkAclBase {
+      public readonly networkAclId = networkAclId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ID of the NetworkACL
+   *
+   * @attribute
+   */
+  public readonly networkAclId: string;
+
+  /**
+   * The VPC ID for this NetworkACL
+   *
+   * @attribute
+   */
+  public readonly networkAclVpcId: string;
+
+  private readonly networkAcl: CfnNetworkAcl;
+  private readonly vpc: IVpc;
+
+  constructor(scope: Construct, id: string, props: NetworkAclProps) {
+    super(scope, id, {
+      physicalName: props.networkAclName
+    });
+
+    this.vpc = props.vpc;
+
+    this.networkAcl = new CfnNetworkAcl(this, 'Resource', {
+      vpcId: props.vpc.vpcId,
+    });
+
+    this.networkAclId = this.networkAcl.ref;
+    this.networkAclVpcId = this.networkAcl.vpcId;
+
+    if (props.subnetSelection !== undefined) {
+      this.associateWithSubnet('DefaultAssociation', props.subnetSelection);
+    }
+  }
+
+  /**
+   * Add a new entry to the ACL
+   */
+  public addEntry(id: string, options: CommonNetworkAclEntryOptions): NetworkAclEntry {
+    return new NetworkAclEntry(this, id, {
+      networkAcl: this,
+      ...options
+    });
+  }
+
+  /**
+   * Associate the ACL with a given set of subnets
+   */
+  public associateWithSubnet(id: string, selection: SubnetSelection) {
+    const subnets = this.vpc.selectSubnets(selection);
```

Comment:
@rix0rrr 
Would this not be better placed under the Subnet Class 
somthing like assocateWithNetworkAcl( id: string ,...) 
The current function his has no way of updating the Subnet's subnetNetworkAclAssociationId
  /**
   * @attribute
   */
  public readonly subnetNetworkAclAssociationId: string;

So if I read back say the publicSubnets , they will be still assoicated with the VPC default that the stack created.

Reply:
Oh wow, this is an interesting problem. There is a `{ Fn::GetAtt: [Subnet, NetworkAclAssociationId]}` intrinsic we're using today, but of course using that is sensitive to timing. It will be the DEFAULT ACL, and not the replaced ACL.

Dang.